modules/core/src/main/scala/com.snowplowanalytics.iglu/client/resolver/ResolverCache.scala

```diff
@@ -118,21 +142,28 @@ object ResolverCache {
         Applicative[F].pure(none)
     }
 
-  def getItem[F[_]: Monad: Clock, A, K](
+  private def getTimestampedItem[F[_]: Monad: Clock, A, K](
     ttl: Option[Int],
     c: LruMap[F, K, (Int, Lookup[A])],
     key: K
-  ): F[Option[Lookup[A]]] = {
+  ): F[Option[(Lookup[A], Int)]] = {
     val wrapped = for {
       (timestamp, lookup) <- OptionT(c.get(key))
       seconds             <- OptionT.liftF(currentSeconds[F])
       useSeconds = seconds
       if isViable(ttl, seconds, timestamp) || lookup.isLeft // isLeft
-    } yield lookup
+    } yield (lookup, timestamp)
```

Comment:
Oh, my favourite type of review comment: `Can we add some case class or type alias for that thing`? 🙂 `TimestampedItem` or sth?

BTW I recall I already added some type aliases in the latest version of client (2.x.x). Is there a plan to apply this feature to the latest version as well?

Reply:
Yes, we definitely need to port it to the latest client!  I worked on the old branch to start with because I wanted to test it out in Enrich.  Migrating it to the CE3 branch can be a task for the next month, but let's agree on a clean API interface first.